tonic/src/transport/service/reconnect.rs

```diff
@@ -89,12 +91,12 @@ where
 
                             state = State::Idle;
 
-                            if self.has_been_connected {
-                                self.error = Some(e.into());
-                                break;
-                            } else {
+                            if !self.has_been_connected && !self.is_lazy {
```

Comment:
```suggestion
                            if !(self.has_been_connected || self.is_lazy) {
```

applying https://en.wikipedia.org/wiki/De_Morgan%27s_laws